Lib/test/test_tcl.py

```diff
@@ -142,7 +142,9 @@ def test_getint(self):
         for i in self.get_integers():
             self.assertEqual(tcl.getint(' %d ' % i), i)
             self.assertEqual(tcl.getint(' %#o ' % i), i)
-            self.assertEqual(tcl.getint((' %#o ' % i).replace('o', '')), i)
+            self.assertEqual(tcl.getint((' %#o ' % i).replace('o', '')),
+                             i if tcl_version < (9, 0) else
```

Comment:
if tcl_version is bigger than 9.0 it is testing the int conversion of tcl.getint() vs python int(), and I don't know if the purpose of the test is that. Just an opinion.

Reply:
As I understand it, the purpose of this test is to verify that Tcl parses various formatted integer strings as expected. In this particular step, `(' %#o ' % i).replace('o', '')` formats `i` as a string in base 8 with a leading 0 prefix, e.g. `2147483647` becomes `'017777777777'`. Tcl prior to 9.0 recognized the leading 0 as meaning base 8, so the expected result from `tcl.getint()` was `i`. But since Tcl 9.0 will now treat leading 0 as base 10, the expected result from `tcl.getint()` is instead what happens when `i` is formatted as base 8 but then parsed as base 10 (e.g. `2147483647` → `'017777777777'`  → `17777777777`), so the equivalent Python code `int(('%#o' % i).replace('o', ''))` is used for the expected result. Maybe there is a better way to improve this test for Tcl 9.0, if it is still useful.

Reply:
(' %#o ' % i).replace('o', '') == ' 0%o ' % i.  No need to add  'o' with '#' and then delete it.